src/main/java/ca/corefacility/bioinformatics/irida/service/impl/sample/SampleServiceImpl.java

```diff
@@ -328,16 +330,30 @@ public SampleSequencingObjectJoin getSampleForSequencingObject(SequencingObject
 	@Override
 	@Transactional(readOnly = true)
 	@PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_SEQUENCER') or hasPermission(#project, 'canReadProject')")
-	public Sample getSampleBySampleName(Project project, String sampleId) {
-		Sample s = sampleRepository.getSampleBySampleName(project, sampleId);
+	public Sample getSampleBySampleName(Project project, String sampleName) {
+		Sample s = sampleRepository.getSampleBySampleName(project, sampleName);
 		if (s != null) {
 			return s;
 		} else {
 			throw new EntityNotFoundException(
-					"No sample with external id [" + sampleId + "] in project [" + project.getId() + "]");
+					"No sample with name [" + sampleName + "] in project [" + project.getId() + "]");
 		}
 	}
 
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	@PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_SEQUENCER') or hasPermission(#project, 'canReadProject')")
+	public Map<String, List<Long>> getSampleIdsBySampleNameForProjects(List<Long> projectIds,
+			List<String> sampleNames) {
+		return sampleRepository.getSampleIdsBySampleNameInProjects(projectIds, sampleNames)
+				.stream()
+				.collect(Collectors.groupingBy(t -> (String) t.get(0),
```

Comment:
Can you rename `t` to `sampleNameIDTuple` so that another contribution looking at this will immediately understand the structure.

Reply:
Please see fb526f3a1b283261c3f4e5ac515e7950ebf945bf.